### ISSUE TITLE
2001 VW New Beetle and Maximum Keys

### ISSUE THREAD

**Issue by schwie** [gmenounos/kw1281test#87]
Greg and team - kw1281test is outstanding. Thanks so much for sharing your time and effort with us!

I am enjoying kw1281test with your recommended cable connected to a 2009 (white) MacBook and a 2001 New Beetle GLS 2.0. Using these components, kw1281test successfully executes the “GetSKC” command and its output matches my Beetle’s scratch-off tag - off to a great start.

Next, I wanted to add a pair of new keys to the Beetle’s immobilizer to go along with two existing keys. The following command also successfully executes (X replaced with my SKC):

./kw1281test 12345678 9600 17 AdaptationSave 21 4 0XXXX

However, no matter what order the four keys are adapted in, the immobilizer only accepts the first three keys inserted into the ignition that are flipped to “On”. After three keys are inserted and switched to “On”, the dash’s security icon flashes indicating it is done programming. When the fourth key is inserted into the Beetle’s ignition the programming mode is already exited. 

When I started typing this description, I wondered if three keys was a kw1281test limitation. Rather, could it be an issue that the 2001 New Beetle is limited to three keys? I also tried modifying the above command by changing the “4” to a “5” and the same issue occurred, a maximum of three keys can be adapted.

When I started this description, I wondered if a cluster reset would help, but I’m guessing that won’t be the case. I’ve tried changing the order of the four keys and any three can work at any one time, but never all four at the same time.

**Comment by CH341A** [comment]
Your're right in only three keys are accepted. VCDS tell everone, there are only three keys maximum. Maybe newer cars can have more then three keys, but for car around the millenium change there are only three accepted.

**Comment by schwie** [comment]
Good to know, I couldn't find this information on the [ross-tech.com support website](https://www.ross-tech.com/vag-com/cars/vwmkIV.html) or really anywhere else. What you say makes sense, I'll proceed with 3 active keys. Thank you!

**Comment by CH341A** [comment]
Yes, is't a bit hidden but within the VW provided service manuals there is a section where a hint to just three keys is mentioned. However, all VWs are limited and also those, connected to VW as Audi, Seat, Skoda and so on. Further it's apparently also a limitation within the dashboard controller. If you buy a new VW at that time you got only two real keys and a third one as an emergency key, just a piece of plastic opening the car by hand.  This third key had a sticker attached, mostly punched characters into sheet metal with the SKU code and is mostly gone by used cars. Hence we need Greg to give us the chance to get the SKU and avoid asking VW.
